### DISCUSSION TITLE
[Next13/appDirectory] Two questions about on-demand build/invalidate everything

### DISCUSSION THREAD

**Discussion by Daavidaviid** [vercel/next.js#43890] [Help] [answered]
Hey,

I'm working a project for client ( It's a website plugged with the Strapi CMS) and we are using **Nextjs13** with **app directory**. Until now, it's working nicely and the dx is great.

And I have **two** questions about on-demand revalidation : 
https://beta.nextjs.org/docs/data-fetching/revalidating#on-demand-revalidation

1. Is it possible to **invalidate everything** ? Like : 
```typescript
try {
    await res.revalidate('*');
    return res.json({ revalidated: true });
  } catch (err) {
    return res.status(500).send('Error revalidating');
  }
```
2. How are we supposed to manage the cases where we **add new content** and **delete existant content** ? According to this screenshot : 
<img width="765" alt="Screenshot 2022-12-09 at 11 10 42" src="https://user-images.githubusercontent.com/32457364/206726355-2c0b9427-6faf-4bb8-a25a-d0d94120a9bd.png">
From https://beta.nextjs.org/docs/api-reference/generate-static-params the new url for new content and deleting the url for previous content won't happen.

So how should we do that ? Do we have an api like : 

```typescript
await res.build('/path-to-build');
await res.erase('/path-to-erase');
```

Or maybe I'm wrong and should have read the documentation completely ? 😬

**Comment by Daavidaviid** [comment, marked as answer]
Hey,

I found the solution. As of today, the solution is to use this undocumented option `unstable_onlyGenerated` : 

<img width="371" alt="Screenshot 2022-12-14 at 13 12 14" src="https://user-images.githubusercontent.com/32457364/207648680-afc7abaf-7c5f-443a-98b4-1132677d4f13.png">

It's working great !! 🥳

**Reply by dvartic** [reply]
Thank you for your contribution. I am also building a website with Next and Strapi and I also needed to remove a route on revalidation. I can confirm that setting `unstable_onlyGenerated: false` works.

**Reply by Daavidaviid** [reply]
> Thank you for your contribution. I am also building a website with Next and Strapi and I also needed to remove a route on revalidation. I can confirm that setting `unstable_onlyGenerated: false` works.

@dvartic did you experience that sometime the nextjs server stop updating ? I'm wondering if it might be linked to nextjs and issues with memory or something wrong on our part

**Reply by dvartic** [reply]
I did not perform extensive testing yet, so I do not know. I will reply again in a couple of days after doing more testing.

**Reply by dvartic** [reply]
Sorry for the late reply. I find revalidation to work fine. What I did find is some sort of caching issue. For example, if I have a post and I change one image for another in Strapi (with another name), the new image will show up fine when opening the page in a new browser (or private window) but will take minutes to show up if I open the page on a browser that has visited it before.

Can you be experiencing something of the sort?

**Reply by karlmacklin** [reply]
> > Thank you for your contribution. I am also building a website with Next and Strapi and I also needed to remove a route on revalidation. I can confirm that setting `unstable_onlyGenerated: false` works.
> 
> @dvartic did you experience that sometime the nextjs server stop updating ? I'm wondering if it might be linked to nextjs and issues with memory or something wrong on our part

Here's a desperate chance of a reply to you Daavidaviid. What you describe there sounds a bit like what we're suffering from right now.

We call res.revalidate() but for some pages nothing happens when calling it, the process just halts. Feels memory related. The rest of our pages work fine. Restarting the server makes these pages work again. The pages are just random, some might stop working for some reason.

We have tried updating from Next 12 to 14 to solve this, no go. We are still on pages router however.

I'm just curious, did you find out anything more about your case where you had the server stop updating?
